File: frontend/api/app/api/v1/trust_defense.py
from fastapi import APIRouter, HTTPException, Path
from app.services.security.trust_defense_engine import TrustDefenseEngine
from app.services.ai_agent_engine import AIFinancialAgent

router = APIRouter(
    prefix="/security",
    tags=["Trust Defense Engine"]
)

@router.get("/trust/{wallet}")
def get_trust_defense(wallet: str = Path(..., description="Wallet address")):
    try:
        engine = TrustDefenseEngine()
        report = engine.generate_defense_report(wallet)
        return {
            "authenticity": report["authenticityScore"],
            "sybilRisk": report["sybilRisk"],
            "trustSafe": report["trustSafe"]
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/report/{wallet}")
def get_ai_defense_report(wallet: str = Path(..., description="Wallet address")):
    try:
        agent = AIFinancialAgent()
        return agent.analyzeTrustAuthenticity(wallet)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: backend/routes/trust_defense.ts
import { Router, Request, Response } from "express";
import { TrustDefenseEngine } from "../services/security/trust_defense_engine";

const router = Router();
const engine = new TrustDefenseEngine();

router.get("/trust/:wallet", (req: Request, res: Response) => {
  const wallet = req.params.wallet;
  const report = engine.generateDefenseReport(wallet);
  res.json({
    authenticity: report.authenticityScore,
    sybilRisk: report.sybilRisk,
    trustSafe: report.trustSafe
  });
});

router.get("/report/:wallet", (req: Request, res: Response) => {
  const wallet = req.params.wallet;
  const report = engine.generateDefenseReport(wallet);
  
  let analysis = "This wallet's reputation appears authentic.";
  if (!report.trustSafe) {
    analysis = "Warning: Possible artificial reputation farming detected.";
  }

  res.json({
    wallet: wallet.toLowerCase(),
    authenticityScore: report.authenticityScore,
    sybilRisk: report.sybilRisk,
    trustSafe: report.trustSafe,
    analysis
  });
});

export default router;
